# ROOT CAUSES

"Root causes" are the fundamental factors or origins that lead to an event or issue. Identifying root causes is crucial for effectively addressing problems and preventing the recurrence of the same issues in the future.

And **SOLUTION** become actions or steps designed to address or overcome the root causes of an issue or event.

### Example

Let's say you have a dataset containing information about customer complaints for a product, and you want to identify the root causes of these complaints.

In [1]:
import pandas as pd

# Sample dataset
data = {
    'Complaint_ID': [1, 2, 3, 4, 5],
    'Complaint_Type': ['Quality Issue', 'Shipping Delay', 'Billing Error', 'Quality Issue', 'Shipping Delay'],
    'Product_Category': ['Electronics', 'Clothing', 'Electronics', 'Electronics', 'Electronics']
}

df = pd.DataFrame(data)

In [2]:
df.head()

Complaint_ID  Complaint_Type Product_Category
0             1   Quality Issue      Electronics
1             2  Shipping Delay         Clothing
2             3   Billing Error      Electronics
3             4   Quality Issue      Electronics
4             5  Shipping Delay      Electronics

In [3]:
# Group complaints by type and product category
complaint_counts = df.groupby(['Complaint_Type', 'Product_Category']).size().reset_index(name='Count')

# Identify root causes by sorting complaints by count
root_causes = complaint_counts.sort_values(by='Count', ascending=False)

print(root_causes)

   Complaint_Type Product_Category  Count
1   Quality Issue      Electronics      2
0   Billing Error      Electronics      1
2  Shipping Delay         Clothing      1
3  Shipping Delay      Electronics      1


In this example:

- We first import the pandas library to work with data in tabular form.
- We create a sample dataset containing complaint IDs, types, and product categories.
- We group the complaints by type and product category using the groupby() function.
- We count the number of complaints for each combination of complaint type and product category.
- Finally, we sort the complaints by count in descending order to identify the most frequent root causes.

# DATA STORYTELLING

Data storytelling in data analytics is the process of presenting data analysis and insights in a narrative format using the Python programming language and its libraries. It involves steps such as data preparation, exploratory data analysis, visualization, as well as communicating the analysis results to stakeholders. The goal is to transform data analysis findings into a compelling story that is engaging and easily understood, thus providing valuable insights and supporting decision-making.

### For Example from a Dataset

In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

##### _This is a dataset about website sephora_

In [4]:
df = pd.read_csv('sephora_website_dataset.csv')

In [3]:
df.head()

id           brand   category                            name  \
0  2218774  Acqua Di Parma  Fragrance  Blu Mediterraneo MINIATURE Set   
1  2044816  Acqua Di Parma    Cologne                         Colonia   
2  1417567  Acqua Di Parma    Perfume                Arancia di Capri   
3  1417617  Acqua Di Parma    Perfume                Mirto di Panarea   
4  2218766  Acqua Di Parma  Fragrance           Colonia Miniature Set   

             size  rating  number_of_reviews  love  price  value_price  ...  \
0  5 x 0.16oz/5mL     4.0                  4  3002   66.0         75.0  ...   
1   0.7 oz/ 20 mL     4.5                 76  2700   66.0         66.0  ...   
2    5 oz/ 148 mL     4.5                 26  2600  180.0        180.0  ...   
3   2.5 oz/ 74 mL     4.5                 23  2900  120.0        120.0  ...   
4  5 x 0.16oz/5mL     3.5                  2   943   72.0         80.0  ...   

  MarketingFlags  MarketingFlags_content  \
0           True             online only   
1           True             online only   
2           True             online only   
3           True             online only   
4           True             online only   

                                             options  \
0                                         no options   
1   - 0.7 oz/ 20 mL  Spray  - 1.7 oz/ 50 mL Eau d...   
2   - 1oz/30mL Eau de Toilette  - 2.5 oz/ 74 mL E...   
3   - 1 oz/ 30 mL Eau de Toilette Spray - 2.5 oz/...   
4                                         no options   

                                             details  \
0  This enchanting set comes in a specially handc...   
1  An elegant timeless scent filled with a fresh-...   
2  Fragrance Family: Fresh Scent Type: Fresh Citr...   
3  Panarea near Sicily is an an island suspended ...   
4  The Colonia Miniature Set comes in an iconic A...   

                                          how_to_use  \
0  Suggested Usage:-Fragrance is intensified by t...   
1                                    no instructions   
2                                    no instructions   
3                                    no instructions   
4  Suggested Usage:-Fragrance is intensified by t...   

                                         ingredients online_only  exclusive  \
0  Arancia di Capri Eau de Toilette: Alcohol Dena...           1          0   
1                                            unknown           1          0   
2  Alcohol Denat.- Water- Fragrance- Limonene- Li...           1          0   
3                                            unknown           1          0   
4  Colonia: Alcohol Denat.- Water- Fragrance- Lim...           1          0   

   limited_edition  limited_time_offer  
0                0                   0  
1                0                   0  
2                0                   0  
3                0                   0  
4                0                   0  

[5 rows x 21 columns]

In [5]:
df.describe()

id       rating  number_of_reviews          love  \
count  9.168000e+03  9168.000000        9168.000000  9.168000e+03   
mean   1.962952e+06     3.990020         282.139180  1.627859e+04   
std    3.859714e+05     1.007707         890.642028  4.260651e+04   
min    5.000000e+01     0.000000           0.000000  0.000000e+00   
25%    1.819453e+06     4.000000          10.000000  1.600000e+03   
50%    2.072354e+06     4.000000          46.000000  4.800000e+03   
75%    2.230591e+06     4.500000         210.000000  1.380000e+04   
max    2.359685e+06     5.000000       19000.000000  1.300000e+06   

             price  value_price  online_only    exclusive  limited_edition  \
count  9168.000000   9168.00000  9168.000000  9168.000000      9168.000000   
mean     50.063237     51.82359     0.234839     0.264725         0.091841   
std      47.164989     49.45902     0.423921     0.441211         0.288817   
min       2.000000      2.00000     0.000000     0.000000         0.000000   
25%      24.000000     25.00000     0.000000     0.000000         0.000000   
50%      35.000000     35.00000     0.000000     0.000000         0.000000   
75%      59.000000     60.00000     0.000000     1.000000         0.000000   
max     549.000000    549.00000     1.000000     1.000000         1.000000   

       limited_time_offer  
count         9168.000000  
mean             0.000327  
std              0.018087  
min              0.000000  
25%              0.000000  
50%              0.000000  
75%              0.000000  
max              1.000000

_Based on the data we obtained from the sephora website, the total sales data contained in the sephora website was 9,168 consumers. From all these data, it can be seen that the average number of reviews given by consumers on the Sephora website is 282 reviews with the average rating obtained from the total product is 3.9 from a rating scale of 5.0. In addition, the average price of products purchased by consumers ranges from 50 dollars._